File: ruby/lib/graphhopper/models/request.rb
module GraphHopper
  # 
  class Request < BaseObject
    attr_accessor :vehicles, :vehicle_types, :services, :shipments, :relations, :algorithm
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # An array of vehicles that can be employed
        :'vehicles' => :'vehicles',
        
        # An array of vehicle types
        :'vehicle_types' => :'vehicle_types',
        
        # An array of services
        :'services' => :'services',
        
        # An array of shipments
        :'shipments' => :'shipments',
        
        # An array of relations
        :'relations' => :'relations',
        
        # 
        :'algorithm' => :'algorithm'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'vehicles' => :'Array<Vehicle>',
        :'vehicle_types' => :'Array<VehicleType>',
        :'services' => :'Array<Service>',
        :'shipments' => :'Array<Shipment>',
        :'relations' => :'Array<Relation>',
        :'algorithm' => :'Algorithm'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'vehicles']
        if (value = attributes[:'vehicles']).is_a?(Array)
          self.vehicles = value
        end
      end
      
      if attributes[:'vehicle_types']
        if (value = attributes[:'vehicle_types']).is_a?(Array)
          self.vehicle_types = value
        end
      end
      
      if attributes[:'services']
        if (value = attributes[:'services']).is_a?(Array)
          self.services = value
        end
      end
      
      if attributes[:'shipments']
        if (value = attributes[:'shipments']).is_a?(Array)
          self.shipments = value
        end
      end
      
      if attributes[:'relations']
        if (value = attributes[:'relations']).is_a?(Array)
          self.relations = value
        end
      end
      
      if attributes[:'algorithm']
        self.algorithm = attributes[:'algorithm']
      end
      
    end

  end
end

File: csharp/src/main/csharp/IO/Swagger/Model/Request.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model
{

    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public class Request :  IEquatable<Request>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="Request" /> class.
        /// </summary>
        public Request()
        {
            
        }

        
        /// <summary>
        /// An array of vehicles that can be employed
        /// </summary>
        /// <value>An array of vehicles that can be employed</value>
        [DataMember(Name="vehicles", EmitDefaultValue=false)]
        public List<Vehicle> Vehicles { get; set; }
  
        
        /// <summary>
        /// An array of vehicle types
        /// </summary>
        /// <value>An array of vehicle types</value>
        [DataMember(Name="vehicle_types", EmitDefaultValue=false)]
        public List<VehicleType> VehicleTypes { get; set; }
  
        
        /// <summary>
        /// An array of services
        /// </summary>
        /// <value>An array of services</value>
        [DataMember(Name="services", EmitDefaultValue=false)]
        public List<Service> Services { get; set; }
  
        
        /// <summary>
        /// An array of shipments
        /// </summary>
        /// <value>An array of shipments</value>
        [DataMember(Name="shipments", EmitDefaultValue=false)]
        public List<Shipment> Shipments { get; set; }
  
        
        /// <summary>
        /// An array of relations
        /// </summary>
        /// <value>An array of relations</value>
        [DataMember(Name="relations", EmitDefaultValue=false)]
        public List<Relation> Relations { get; set; }
  
        
        /// <summary>
        /// Gets or Sets Algorithm
        /// </summary>
        [DataMember(Name="algorithm", EmitDefaultValue=false)]
        public Algorithm Algorithm { get; set; }
  
        
  
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class Request {\n");
            sb.Append("  Vehicles: ").Append(Vehicles).Append("\n");
            sb.Append("  VehicleTypes: ").Append(VehicleTypes).Append("\n");
            sb.Append("  Services: ").Append(Services).Append("\n");
            sb.Append("  Shipments: ").Append(Shipments).Append("\n");
            sb.Append("  Relations: ").Append(Relations).Append("\n");
            sb.Append("  Algorithm: ").Append(Algorithm).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as Request);
        }

        /// <summary>
        /// Returns true if Request instances are equal
        /// </summary>
        /// <param name="obj">Instance of Request to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(Request other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.Vehicles == other.Vehicles ||
                    this.Vehicles != null &&
                    this.Vehicles.SequenceEqual(other.Vehicles)
                ) && 
                (
                    this.VehicleTypes == other.VehicleTypes ||
                    this.VehicleTypes != null &&
                    this.VehicleTypes.SequenceEqual(other.VehicleTypes)
                ) && 
                (
                    this.Services == other.Services ||
                    this.Services != null &&
                    this.Services.SequenceEqual(other.Services)
                ) && 
                (
                    this.Shipments == other.Shipments ||
                    this.Shipments != null &&
                    this.Shipments.SequenceEqual(other.Shipments)
                ) && 
                (
                    this.Relations == other.Relations ||
                    this.Relations != null &&
                    this.Relations.SequenceEqual(other.Relations)
                ) && 
                (
                    this.Algorithm == other.Algorithm ||
                    this.Algorithm != null &&
                    this.Algorithm.Equals(other.Algorithm)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.Vehicles != null)
                    hash = hash * 57 + this.Vehicles.GetHashCode();
                
                if (this.VehicleTypes != null)
                    hash = hash * 57 + this.VehicleTypes.GetHashCode();
                
                if (this.Services != null)
                    hash = hash * 57 + this.Services.GetHashCode();
                
                if (this.Shipments != null)
                    hash = hash * 57 + this.Shipments.GetHashCode();
                
                if (this.Relations != null)
                    hash = hash * 57 + this.Relations.GetHashCode();
                
                if (this.Algorithm != null)
                    hash = hash * 57 + this.Algorithm.GetHashCode();
                
                return hash;
            }
        }

    }
}
